### ISSUE TITLE
Linker error when building fmusim

### ISSUE THREAD

**Issue by malchera** [modelica/Reference-FMUs#360]
Hi there, I get a linker error when trying to build the `fmusim` target.
Version is:
```
$ git describe
v0.0.23
```

Steps to reproduce:
* Run `python3 build.py` in repository root directory (same error as below)

Alternatively:
* Run CMake manually:
```
mkdir build && cd build
cmake .. -DFMI_TYPE=CS -DFMI_VERSION=3 -DWITH_FMUSIM=ON
cmake --build .
```
This compiles fine but gives a linker error in the end:
```
[...]
[ 87%] Linking C executable fmusim
/usr/bin/ld: CMakeFiles/fmusim.dir/FMICVode.c.o:(.bss+0x0): multiple definition of `SolverImpl_'; CMakeFiles/fmusim.dir/FMIEuler.c.o:(.bss+0x0): first defined here
collect2: error: ld returned 1 exit status
gmake[2]: *** [fmusim/CMakeFiles/fmusim.dir/build.make:436: fmusim/fmusim] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:910: fmusim/CMakeFiles/fmusim.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2
```

This happens with all combinations of types/versions `{CS, ME} x {1, 2, 3}` when `fmusim` is enabled.

Moreover, the "Build the fmus" description in `README.md` could be improved by mentioning that fmusim requires all Python scripts in `ThirdParty` to be run once before CMake, such that all dependencies are built. Otherwise there are compilation errors, because the CMake file assumes a couple of paths to cvode, libcml and zlib which do not exist.

**Comment by malchera** [comment]
On a different note (sorry if a bit off-topic): Are there any resources available where it is explained how to create a FMU (XML, C code) _from scratch_? I seem unable to find it. All resources I've found so far are either very high level, guide you through the FMI implementation of tool xyz or they focus on particular parts of a bigger code. I'd simple like to know how to get started using my bare metal machine.

In other words, what I would like to achieve is to start off with an empty project and create a (CS) FMU from scratch that simply exchanges data (e.g., send an integer from A to B). I would be grateful if you could point me to some resources on that. Thank you!

**Comment by t-sommer** [comment]
The easiest way to build the FMUs for all FMI versions and fmusim is to run the [build scripts from CI](https://github.com/modelica/Reference-FMUs/blob/main/.github/workflows/build.yml#L39).

**Comment by t-sommer** [comment]
>Are there any resources available where it is explained how to create a FMU (XML, C code) _from scratch_?

Please ask this question on [StackOverflow](https://stackoverflow.com/).

**Comment by miaoruonan** [comment]
只需修改fmusim/FMIEuler.c和FMICVode.c中修改相同定义的SolverImpl_结构体名字和引用即可
